### DISCUSSION TITLE
Beta.20 MOAR SPEED! ⚡️

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#330] [Announcements]
## Process Pools and Virtualized Lists

https://user-images.githubusercontent.com/36073/123519955-7fdd8200-d66b-11eb-8167-09b9daed1c9f.mp4

## Experimental `textarea`

Feel free to play around with the `textarea` for multiline input.

```js
let value = await textarea()
```

The API of textarea will change (it currently just sets the placeholder), but it will always return the string value of the textarea, so there won't be any breaking changes if you just keep the default behavior. `cmd+s` submits. `cmd+w` cancels.

## Experimental `editor` (this will become a _paid_  💵 feature later this year)

As an upgrade to `textarea`, `await editor()` will give you a full editor experience. Same as the textarea, the API will also change, but will always return a string of the content.

```js
// Defaults to markdown
let value = await editor()
```

> ⚠️ API is subject to change!
```js
let value = await editor("markdown", `
## Preloaded content

* nice
`)
```

```js
let value = await editor("javascript", `
console.log("Support other languages")
`)
```

### A note on paid features

Everything you've used so far in the Script Kit app will stay free. The core `kit` is open-source MIT. 

The paid features will be add-ons to the core experience: Themes, Editor, Widgets, Screenshots, Record Audio, and many more fun ideas. These will roll out experimentally in the free version first then move exclusively to the paid version. Expect the paid versions later this year.

**Comment by TreTuna** [comment]
This has been looking so good @johnlindquist! Textarea has been working great for my quick notes script.

I look forward to being one of your first paid customers!

**Reply by johnlindquist** [reply]
Thanks @TreTuna! Had a couple of great breakthroughs these past couple weeks and I was super excited to get this build out. It's turned out so great and I'm  super proud of it.

I can't wait for you to spend money on it 😇

**Comment by Alarid** [comment]
Wow, the speed improvement is massive, thanks a lot @johnlindquist !

**Comment by vojtaholik** [comment]
Loving the speed improvements, amazing work @johnlindquist!
